realm/kotlin-extensions/src/androidTest/kotlin/io/realm/KotlinRealmTests.kt

```diff
@@ -61,4 +66,88 @@ class KotlinRealmTests {
     fun where() {
         assertEquals(0, realm.where<SimpleClass>().count())
     }
-}
+
+    @Test
+    fun where_getsResult() {
+        assertEquals(0, realm.where<PrimaryKeyClass>().count())
+
+        val pk = 1L
+        val name = "complex"
+
+        realm.executeTransaction {
+            val pkClassObj = it.createObject(PrimaryKeyClass::class.java, pk)
+            pkClassObj.name = name
+
+        }
+
+        val pkClassObjOut = realm.where<PrimaryKeyClass>()
+                .equalTo(PrimaryKeyClass::id.name, pk)
+                .findFirst()
+
+        assertNotNull(pkClassObjOut)
+        assertEquals(pk, pkClassObjOut?.id)
+        assertEquals(name, pkClassObjOut?.name)
+    }
+
+    @Test
+    fun callInTransaction_withPairOfRealmModelReturned() {
```

Comment:
I am not sure I understand the value of the `callInTransaction` extension function?

Is it to return the object being created? Is that use case really important enough to warrant an extension function in our official API?

Also, I'm not sure about the name either? Is it a naming convention from Kotlin?